### PR TITLE
Mark Environment.OSVersion as deprecated

### LINKED ISSUE #58: Mark Environment.OSVersion as deprecated
`Environment.OSVersion` has issues, for starters it doesn't return a sensible value for macOS:

```C#
// Output:
//
//      macOS 10.12.6 (16G29)    --> Hello from Unix (Unix 16.7.0.0)
//      Ubuntu 16.04.3 LTS       --> Hello from Unix (Unix 4.4.0.43)
//      Windows 1709 (16299.19)  --> Hello from Win32NT (Microsoft Windows NT 6.2.9200.0)

Console.WriteLine($"Hello from {Environment.OSVersion.Platform} ({Environment.OSVersion})");
```

But as per @weshaggard there are other issues such as:

1. People checking for exact enum values for PlatformId. That break things for Unix vs Linux vs OSX which value should I check for? I think our existing IsOSPlatform allows for better checking
1. Also the enum value for the platform checks is nearly impossible to update in all our platforms as we add new ones. This is why OSPlatform can be default to a string until we can make the constant available in all our platforms. 
1. We cannot actually make the enum value different without breaking compat so things like OSX will never be returned as a platform as an example. 
1. Too much code does exact version checks instead of IsAtLeast this version type of check. We have yet to solve this.
1. OS’s like windows start to lie about the OSVersion because of people doing exact version checks so then folks really don’t have a value to query.  We can help with this with better version check APIs. 
1. We want to help folks find other ways to solve the scenarios they would use version for (i.e. feature light-up) to help eliminate some of the need to do version checks at all.

### PR DESCRIPTION
**Pull request by pjanotti**
Fixes #58

### CODE REVIEW

File: docs/DE0009.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<!--
```
Comment (pjanotti) [inline docs/DE0009.md]: @mairaw could you please review this text?

File: docs/DE0009.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+P:System.Environment.OSVersion
+-->
+
+# DE0009: Environment.OSVersion shouldn't be used
+
+## Motivation
+
+The `OSVersion` property doesn't provide a reliable way to identify the exact OS and its version.
```
Comment (mairaw) [inline docs/DE0009.md]: should we link to the APIs?

File: docs/DE0009.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+## Recommendation
+
+Use the method `RuntimeInformation.IsOSPlatform` to identify the OS platform and avoid writing
```
Comment (mairaw) [inline docs/DE0009.md]: We usually say what the API is after: Use the `RuntimeInformation.IsOSPlatform` method...

Link to the API on docs? (https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.runtimeinformation.isosplatform)

The last part of this sentence is a bit hard to read. How about this?
Use the `RuntimeInformation.IsOSPlatform` method to identify the OS platform. Avoid writing code dependent on reported OS version and instead check for availability of needed features (ideally you'd say how here).

Comment (pjanotti) [reply docs/DE0009.md]: The exact check depends on the feature in question so here it needs to be very generic.

Comment (mairaw) [reply docs/DE0009.md]: That's ok. Just to be super clear, I was more thinking of the technique you'd use to do the check. But  if you can't describe it, it's fine as-is.

**Comment by pjanotti** [comment]
Thanks for the review @mairaw

**Pull request merged**